RecoBTag/ONNXRuntime/plugins/ParticleNetSonicJetTagsProducer.cc

```diff
@@ -0,0 +1,317 @@
+    input_names_ = prep_pset.getParameter<std::vector<std::string>>("input_names");
+    for (const auto &group_name : input_names_) {
+      const auto &group_pset = prep_pset.getParameterSet(group_name);
+      auto &prep_params = prep_info_map_[group_name];
+      prep_params.var_names = group_pset.getParameter<std::vector<std::string>>("var_names");
+      prep_params.min_length = group_pset.getParameter<unsigned>("var_length");
+      prep_params.max_length = prep_params.min_length;
+      const auto &var_info_pset = group_pset.getParameterSet("var_infos");
+      for (const auto &var_name : prep_params.var_names) {
+        const auto &var_pset = var_info_pset.getParameterSet(var_name);
+        double median = var_pset.getParameter<double>("median");
+        double norm_factor = var_pset.getParameter<double>("norm_factor");
+        double replace_inf_value = var_pset.getParameter<double>("replace_inf_value");
+        double lower_bound = var_pset.getParameter<double>("lower_bound");
+        double upper_bound = var_pset.getParameter<double>("upper_bound");
+        prep_params.var_info_map[var_name] =
+            PreprocessParams::VarInfo(median, norm_factor, replace_inf_value, lower_bound, upper_bound, 0);
+      }
+    }
+  }
+  if (debug_) {
+    for (unsigned i = 0; i < input_names_.size(); ++i) {
+      const auto &group_name = input_names_.at(i);
+      if (!input_shapes_.empty()) {
+        std::cout << group_name << "\nshapes: ";
+        for (const auto &x : input_shapes_.at(i)) {
+          std::cout << x << ", ";
+        }
+      }
+      std::cout << "\nvariables: ";
+      for (const auto &x : prep_info_map_.at(group_name).var_names) {
+        std::cout << x << ", ";
+      }
+      std::cout << "\n";
+    }
+    std::cout << "flav_names: ";
+    for (const auto &flav_name : flav_names_) {
+      std::cout << flav_name << ", ";
+    }
+    std::cout << "\n";
+  }
+
+  // get output names from flav_names
+  for (const auto &flav_name : flav_names_) {
+    produces<JetTagCollection>(flav_name);
+  }
+}
+
+ParticleNetSonicJetTagsProducer::~ParticleNetSonicJetTagsProducer() {}
+
+void ParticleNetSonicJetTagsProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  // pfDeepBoostedJetTags
+  edm::ParameterSetDescription desc;
+  TritonClient::fillPSetDescription(desc);
+  desc.add<unsigned>("batchSize", 1);
+  desc.add<edm::InputTag>("src", edm::InputTag("pfDeepBoostedJetTagInfos"));
+  desc.add<std::string>("preprocess_json", "");
+  // `preprocessParams` is deprecated -- use the preprocessing json file instead
+  edm::ParameterSetDescription preprocessParams;
+  preprocessParams.setAllowAnything();
+  preprocessParams.setComment("`preprocessParams` is deprecated, please use `preprocess_json` instead.");
+  desc.addOptional<edm::ParameterSetDescription>("preprocessParams", preprocessParams);
+  desc.add<std::vector<std::string>>("flav_names",
```

Comment:
better to obtain this directly from `HeterogeneousCore/SonicTriton/data/models/particlenet/particlenet_labels.txt` to ensure consistency and reduce duplication. For an example, see:
https://github.com/cms-sw/cmssw/blob/763e00fdf07f4a4c00730b139d52ed5113a3b99c/HeterogeneousCore/SonicTriton/test/tritonTest_cfg.py#L103
https://github.com/cms-sw/cmssw/blob/763e00fdf07f4a4c00730b139d52ed5113a3b99c/HeterogeneousCore/SonicTriton/test/TritonImageProducer.cc#L22-L32

Reply:
Hi Kevin, this is another one that I'm not totally sure how to implement.  The strings in HeterogeneousCore/SonicTriton/data/models/particlenet/particlenet_labels.txt are actually different from the ones that get assigned within RecoBTag/ONNXRuntime/plugins/ParticleNetSonicJetTagsProducer.cc.  Should I put the ones that are currently in ParticleNetSonicJetTagsProducer.cc into particlenet_labels.txt?

Reply:
@wpmccormack @kpedro88, there are 3 places where the names are specified and need to be consistent currently:

in `particlenet_labels.txt`: https://github.com/fastmachinelearning/sonic-models/pull/2/files#diff-2eec9163f63ecec5b95df2a4d700eac0dcfbee087777e0824a4772630806ddf3R1-R20

in `preprocess_noragged.json`: https://github.com/jmduarte/RecoBTag-Combined/blob/noragged/ParticleNetAK8/General/V01/preprocess_noragged.json#L2-L22

and in `pfParticleNet_cff.py`: https://github.com/fastmachinelearning/cmssw/pull/3/files#diff-f47cec1537d460518348a7f49589c83fa86f13fdbaab330c9b6687116103b388R37-R39

I agree we can just specify them in one place (ideally in `preprocess_noragged.json` to be consistent with the existing model or in `particlenet_labels.txt` to be consistent with other triton models), and the plugin can just reuse that for `flav_names`

Reply:
I realize now that this won't work because `fillDescriptions()` is a static function, and therefore can't access class member data like the location of `particlenet_labels.txt`. In that case, I would suggest leaving the default value as an empty vector, and get the values in the `cff` file from a common source; this at least reduces duplication somewhat...

Reply:
Hi @jmduarte and @kpedro88 , when you say the names need to be consistent, that means that the strings should be the same, right?  The list of strings right now is different in particle_labes.txt, preprocess_noragged.json, pfParticleNet_cff.py, and ParticleNetSonicJetTagsProducer.cc.  Which one is right?  Or by consistent, do you just mean that the ordering has to be the same.  ParticleNetSonicJetTagsProducer.cc right now has a shorter list than the others, so it's definitely inconsistent right now